src/pages/progress/index.tsx

```diff
@@ -48,26 +53,101 @@ const WarningSection: FC = () => (
   </section>
 );
 
-const ProgressContent: FC<{ currentPhaseIndex: number }> = ({ currentPhaseIndex }) => (
-  <Box className="flex flex-col items-center justify-center mt-4">
-    <div className="flex flex-col items-center justify-center max-w-[400px]">
-      <span className="w-10 text-blue-700 loading loading-spinner"></span>
-      <h1 className="my-3 text-xl font-bold text-blue-700">Your transaction is in progress.</h1>
-      <WarningSection />
-      <div className="h-0.5 m-auto w-full bg-blue-700 mt-3 mb-1" />
-      <ProgressSteps currentPhaseIndex={currentPhaseIndex} />
-    </div>
-  </Box>
-);
+const ProgressContent: FC<{ currentPhase: OfframpingPhase; currentPhaseIndex: number }> = ({
+  currentPhase,
+  currentPhaseIndex,
+}) => {
+  const [currentPercentage, setCurrentPercentage] = useState<number>(
+    Math.round((100 / numberOfPhases) * currentPhaseIndex),
+  );
+  const [targetPhaseIndex, setTargetPhaseIndex] = useState<number>(currentPhaseIndex);
+  const currentPhaseRef = useRef<OfframpingPhase>(currentPhase);
```

Comment:
I had to define some variables as refs instead of state so that the interval that is created inside the `useEffect` would not need to be recreated whenever anyone of the variables changes. This way the `useEffect` hook has an empty dependency array.

Reply:
Only the `currentPercentage` and `targetPhaseIndex` are state variables because changing them needs to trigger a rerender (either show another percentage or start a new transition animation).